src/Route/DeferredRoute.php

```diff
@@ -112,11 +110,16 @@ private function normalizeParameters (array $parameters) : array
 
 
     /**
-     * @param self|string|mixed|null $value
+     * @param self|string|null $value
      */
-    public static function generateValue ($value, UrlGeneratorInterface $urlGenerator) : string
+    public static function generateValue ($value, UrlGeneratorInterface $urlGenerator) : ?string
     {
-        if (\is_string($value) || null === $value)
+        if (null === $value)
+        {
+            return null;
+        }
+
+        if (\is_string($value))
         {
             return (string) $value;
         }
```

Comment:
Why not just do something like this? We don't need to cast if it's already a string.

```suggesstion
        if (\is_string($value) || null === $value)
        {
            return $value;
        }
```

Reply:
Well... yes.

Reply:
That's what you get for not cleaning up your code when refactoring 👍